### ISSUE TITLE
additionalItems should only apply for tuples

### ISSUE THREAD

**Issue by jonaslagoni** [asyncapi/modelina#1138]
#### Describe the bug
Currently, additionalItems are applied even when the items are not a tuple. 

https://datatracker.ietf.org/doc/html/draft-handrews-json-schema-validation-01#section-6.4.2

This means that this:

```
    "tags": {
      "type": "array",
      "items": {
        "$ref": "http://asyncapi.com/definitions/2.6.0/tag.json"
      },
      "uniqueItems": true
    },
```
generates this union type:
```
private _tags?: (Tag | any)[];
```

When it should only be:

```
private _tags?: Tag[];
```

**Comment by jonaslagoni** [comment]
The simple array type above makes sense.

But what about those tuples... Say we have this:
```
    "tags": {
      "type": "array",
      "items": [{type: string}, {type: boolean}],
      "additionalItems": true
    },
```

Would we expect to generate:
```
private _tags?: (string | boolean | any)[];
```

or rather a tuple and an array type in union 🤔
```
private _tags?: [string, boolean] | (string | boolean | any)[];
```

I am leaning towards the last option because the following input is valid:
```
{
  tags: ["teest", true, "test"]
}
```
Same with:
```
{
  tags: ["teest", true, false]
}
```
Same with:
```
{
  tags: ["teest", true]
}
```
And in most languages its either a tuple or array, it cant be partly tuple and partly array. So even though you lose the tuple structure for the array type `(string | boolean | any)[]` it's the best possible type to generate 🤔

**Comment by jonaslagoni** [comment]
Continued with tuple discussion here: https://github.com/asyncapi/modelina/issues/1139

**Comment by kennethaasan** [comment]
I've had issues with differences between draft 7 and draft 2020-12: https://ajv.js.org/json-schema.html#draft-2020-12
But I guess that's a different issue and should be handled by checking draft version

**Comment by jonaslagoni** [comment]
Gonna be solved in `v2` and currently released as part of `2.0.0-next.4`
